benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java

```diff
@@ -493,10 +525,15 @@ private TestResult generateLoad(List<BenchmarkProducer> producers, long testDura
     }
 
     @Override
-    public void messageReceived(byte[] data) {
+    public void messageReceived(byte[] data, long publishTimestamp) {
         messagesReceived.increment();
         totalMessagesReceived.increment();
         bytesReceived.add(data.length);
+
+        long now = System.currentTimeMillis();
+        long endToEndLatencyMillis = now - publishTimestamp;
```

Comment:
This will work; my impl allows for an independent consumer to pull the publish time b/c it is in embedded in the message rather than passed to the callback.

Reply:
Also in this case the publish latency is embedded in the message (both Pulsar & Kafka have the timestamp field)